src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -549,23 +601,63 @@ function tryRemoveElement(elem) {
         /**
          * @private
          */
-        updateCurrentTrackOffset(offsetValue) {
+        updateCurrentTrackOffset(offsetValue, currentTrackIndex = PRIMARY_TEXT_TRACK_INDEX) {
+            let offsetToCompare = this.#currentTrackOffset;
+            if (this.isSecondaryTrack(currentTrackIndex)) {
+                offsetToCompare = this.#secondaryTrackOffset;
+            }
+
             let relativeOffset = offsetValue;
             const newTrackOffset = offsetValue;
-            if (this.#currentTrackOffset) {
-                relativeOffset -= this.#currentTrackOffset;
+
+            if (offsetToCompare) {
+                relativeOffset -= offsetToCompare;
+            }
+
+            if (this.isSecondaryTrack(currentTrackIndex)) {
+                this.#secondaryTrackOffset = newTrackOffset;
+            } else {
+                this.#currentTrackOffset = newTrackOffset;
             }
-            this.#currentTrackOffset = newTrackOffset;
+
             // relative to currentTrackOffset
             return relativeOffset;
         }
 
+        /**
+         * @private
+         * These browsers will not clear the existing active cue when setting an offset
+         * for native TextTracks.
+         * Any previous text tracks that are on the screen when the offset changes will
+         * remain next to the new tracks until they reach the new offset's instance of the track.
+         */
+        requiresHidingActiveCuesOnOffsetChange() {
+            return !!browser.firefox;
+        }
+
+        /**
+         * @private
+         */
+        hideTextTrackActiveCues(currentTrack) {
+            if (currentTrack.activeCues) {
+                Array.from(currentTrack.activeCues).forEach((cue) => {
+                    cue.text = '';
+                });
+            }
+        }
```

Comment:
@dmitrylyzo @thornbill I tried this again and it actually might not be as big of a problem as I originally thought.
I cannot reproduce this in firefox with my prod 10.8.9 on my home server. But I can confirm that I *can* still reproduce this in firefox (chrome is fine) for the current `master` branch (10.9.0) debug build via `npm start`.

The screenshot below was from me quickly changing the offset multiple times. You can see that the track cues have stacked up and are stuck. Again, this seems to not happen in the prod build. 
It only seems to happen on firefox when served via the `start` script.

So maybe this fix is only actually relevant when debugging in the first place.

<img width="616" alt="Screen Shot 2023-02-09 at 7 36 08 PM" src="https://user-images.githubusercontent.com/30599893/217972440-c09c08bf-22b6-4a95-a6c8-8ef771188030.png">

Reply:
Hmm, this issue occurs even with the current fix.
I tried changing `track.mode` to `disabled`, then `showing` to remove the extra lines.

I think that if it happens in `devel` and doesn't happen in `prod`, it's just luck. :smile:

Reply:
Are you able to reproduce it consistently even with this fix? I'm not able to reproduce it at all still.

Does disabling then enabling fix it for you?
Like this?
```
hideTextTrackActiveCues(currentTrack) {
      currentTrack.mode = 'disabled';
      currentTrack.mode = 'showing';
}
```

Reply:
Disabling then showing like I mentioned above doesn't seem to work for me. I am able to reproduce the error after making the change I mentioned above.

The initial fix of clearing the cue with empty strings still works for me and I cannot reproduce the issue when the cues are blanked out

Reply:
> Are you able to reproduce it consistently even with this fix?

Yes. I just open the slider and click +2..3 seconds. Clicking back (0..+1 seconds) clears them though.

Clearing them after an offset (move `hideTextTrackActiveCues` below the offset block) works, but they are not displayed when paused.

:warning: This fix seems to corrupt the cues. Subtitles are not displayed when rewinding.

> Disabling then showing like I mentioned above doesn't seem to work for me.

I had to change it like this because `currentTrack.cues` became `null` after disabling the track:
```diff
diff --git a/src/plugins/htmlVideoPlayer/plugin.js b/src/plugins/htmlVideoPlayer/plugin.js
index 700eedd14..a86b32d4f 100644
--- a/src/plugins/htmlVideoPlayer/plugin.js
+++ b/src/plugins/htmlVideoPlayer/plugin.js
@@ -639,11 +639,11 @@ function tryRemoveElement(elem) {
          * @private
          */
         hideTextTrackActiveCues(currentTrack) {
-            if (currentTrack.activeCues) {
-                Array.from(currentTrack.activeCues).forEach((cue) => {
-                    cue.text = '';
-                });
-            }
+            currentTrack.mode = 'disabled';
+
+            setTimeout(() => {
+                currentTrack.mode = 'showing';
+            }, 0);
         }
 
         /**
@@ -655,14 +655,15 @@ function tryRemoveElement(elem) {
                 if (offsetValue === 0) {
                     return;
                 }
-                if (this.requiresHidingActiveCuesOnOffsetChange()) {
-                    this.hideTextTrackActiveCues(currentTrack);
-                }
                 Array.from(currentTrack.cues)
                     .forEach(function (cue) {
                         cue.startTime -= offsetValue;
                         cue.endTime -= offsetValue;
                     });
+
+                if (this.requiresHidingActiveCuesOnOffsetChange()) {
+                    this.hideTextTrackActiveCues(currentTrack);
+                }
             }
         }
 
```
But the timeout doesn't allow the subtitle offset slider to change the offset when rapidly clicking - it falls here because all tracks are disabled (I'm testing with one subtitle track):
https://github.com/jellyfin/jellyfin-web/blob/276a59191bdc0715e61613e22f2f647d22bc34e3/src/plugins/htmlVideoPlayer/plugin.js#L596

Reply:
I also tried with a 0ms timeout and was running into the same issue. 

Hmmm, I will also look at this a bit more.

Since this is trying to fix an issue that is currently unrelated to the dual subtitles and a bit out of scope of this PR, do you think I should just remove this (not fully working) fix from this PR and create a new separate one specifically for this issue?

Reply:
> Since this is trying to fix an issue that is currently unrelated to the dual subtitles and a bit out of scope of this PR, do you think I should just remove this (not fully working) fix from this PR and create a new separate one specifically for this issue?

Yes, I think it would be better to make another PR for that. Moreover, we can try to fix this in 10.8 (if possible).

Reply:
I've opened a draft PR #4349. I will continue my work on this bug there.